web/users.go

```diff
@@ -52,12 +52,17 @@ func (s *AutograderService) getUsers() (*pb.Users, error) {
 // updateUser promotes the given user to administrator and/or
 // make other changes to the user database entry for the given user.
 // The isAdmin flag must be true to promote the given user to admin.
-func (s *AutograderService) updateUser(isAdmin bool, request *pb.User) (*pb.User, error) {
+func (s *AutograderService) updateUser(isAdmin bool, curUser string, request *pb.User) (*pb.User, error) {
```

Comment:
Change signature take `currentUser` (`*pb.User` object) instead. Then replace the `isAdmin` bool and `curUser` string with `currentUser.IsAdmin` and `currentUser.Login`.